terraso_backend/apps/soil_id/graphql/soil_id.py

```diff
@@ -201,26 +203,84 @@ class SoilIdInputData(graphene.InputObjectType):
 ]
 
 
-# to be replaced by actual algorithm output
-def resolve_location_based_soil_matches(_parent, _info, latitude: float, longitude: float):
-    return LocationBasedSoilMatches(
-        matches=[
-            LocationBasedSoilMatch(
-                data_source="SSURGO",
-                distance_to_nearest_map_unit_m=0.0,
-                match=SoilMatchInfo(score=1.0, rank=0),
-                soil_info=sample_soil_infos[0],
+def resolve_rock_fragment_volume(rock_fragment_volume: int):
+    if rock_fragment_volume <= 1:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_0_1
+    elif rock_fragment_volume <= 15:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_1_15
+    elif rock_fragment_volume <= 35:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_15_35
+    elif rock_fragment_volume <= 60:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_35_60
+    else:
+        return DepthDependentSoilData.RockFragmentVolume.VOLUME_60
+
+
+def resolve_soil_data(soil_match):
+    bottom_depths = soil_match["bottom_depth"]
+    prev_depth = 0
+    depth_dependent_data = [None] * len(bottom_depths)
+
+    for id, bottom_depth in bottom_depths.items():
+        depth_dependent_data[int(id)] = SoilIdDepthDependentData(
+            depth_interval=DepthInterval(start=prev_depth, end=bottom_depth),
+            texture=soil_match["texture"][id].upper().replace(" ", "_"),
+            rock_fragment_volume=resolve_rock_fragment_volume(soil_match["rock_fragments"][id]),
+            munsell_color_string=soil_match["munsell"][id],
+        )
+
+    return SoilIdSoilData(
+        slope=soil_match["site"]["siteData"]["slope"], depth_dependent_data=depth_dependent_data
+    )
+
+
+def resolve_soil_match(soil_match):
+    soil_id = soil_match["id"]
+    site_data = soil_match["site"]["siteData"]
+    ecological_site = soil_match["esd"]["ESD"]
+    if ecological_site["ecoclassid"] == "":
+        ecological_site = None
+    else:
+        ecological_site = EcologicalSite(
+            name=ecological_site["ecoclassname"],
+            id=ecological_site["ecoclassid"],
+            url=ecological_site["esd_url"],
+        )
```

Comment:
```suggestion
    if ecological_site["ecoclassid"]:
        ecological_site = EcologicalSite(
            name=ecological_site["ecoclassname"],
            id=ecological_site["ecoclassid"],
            url=ecological_site["esd_url"],
        )
    else:
        ecological_site = None
```

Reply:
i prefer being explicit about what i'm checking for in conditions